src/steamship/agents/service/agent_service.py

```diff
@@ -79,3 +83,51 @@ def run_agent(self, agent: Agent, context: AgentContext):
         for func in context.emit_funcs:
             logging.info(f"Emitting via function: {func.__name__}")
             func(action.output, context.metadata)
+
+    @post("prompt")
+    def prompt(self, prompt: str, **kwargs) -> List[Block]:
+        """Run an agent with the provided text as the input."""
+
+        # AgentContexts serve to allow the AgentService to run agents
+        # with appropriate information about the desired tasking.
+        # Here, we create a new context on each prompt, and append the
+        # prompt to the message history stored in the context.
+        context_id = uuid.uuid4()
+        context = AgentContext.get_or_create(self.client, {"id": f"{context_id}"})
+        context.chat_history.append_user_message(prompt)
+
+        # Add a default LLM
+        context = with_llm(context=context, llm=OpenAI(client=self.client, model_name="gpt-4-0613"))
+
+        # AgentServices provide an emit function hook to access the output of running
+        # agents and tools. The emit functions fire at after the supplied agent emits
+        # a "FinishAction".
+        #
+        # Here, we show one way of accessing the output in a synchronous fashion. An
+        # alternative way would be to access the final Action in the `context.completed_steps`
+        # after the call to `run_agent()`.
+        output_blocks = []
+
+        def sync_emit(blocks: List[Block], meta: Metadata):
+            nonlocal output_blocks
+            output_blocks.extend(blocks)
+
+        context.emit_funcs.append(sync_emit)
+
+        # Get the agent
+        agent: Optional[Agent] = None
+        if hasattr(self, "agent"):
+            agent = self.agent
+        elif hasattr(self, "_agent"):
+            agent = self._agent
+
+        if not agent:
+            raise SteamshipError(
+                message="No Agent object found in the Agent Service. "
+                "Please name it either self.agent or self._agent."
+            )
+
+        self.run_agent(agent, context)
```

Comment:
I _feel_ like this is missing the `context.chat_history.append_assistant_message(output)` bit. Without that, the chat history will be missing all AI responses.

Reply:
Done!